WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -620,6 +654,35 @@ private extension ProductFormViewController {
         SharingHelper.shareURL(url: url, title: product.name, from: view, in: self)
     }
 
+    func displayDeleteProduct() {
+        presentProductConfirmationDeleteAlert { [weak self] (confirm) in
+            guard confirm else {
+                return
+            }
+
+            self?.viewModel.deleteProductRemotely { [weak self] result in
+                guard let self = self else {
+                    return
+                }
+                switch result {
+                case .failure(let error):
+                    DDLogError("⛔️ Error deleting Product: \(error)")
+                    CrashLogging.logError(error)
+                    // Presents the error alert.
+                    self.displayError(error: error)
+                case .success:
+                    // Pop to the previous view controller
+                    switch self.presentationStyle {
+                    case .navigationStack:
+                        self.navigationController?.popViewController(animated: true)
+                    default:
+                        self.dismiss(animated: true, completion: nil)
+                    }
```

Comment:
nit: maybe these could be moved to a separate function for potential reuse in the future whenever we want to dismiss the product form? then the comment can be replaced by the function name :)

Reply:
Make sense 👍 updated here c0119edeea4afab72a994635c392358766599597 (together with other nits)